Lib/_markupbase.py

```diff
@@ -9,7 +9,7 @@
 
 _declname_match = re.compile(r'[a-zA-Z][-_.a-zA-Z0-9]*\s*').match
 _declstringlit_match = re.compile(r'(\'[^\']*\'|"[^"]*")\s*').match
-_commentclose = re.compile(r'--\s*>')
+_commentclose = re.compile(r'--!?>')
```

Comment:
I would leave the `\s*`, even though I should double check what the HTML5 specs say exactly.

Reply:
> I would leave the `\s*`, even though I should double check what the HTML5 specs say exactly.

I provided the links earlier and "\s*" mentioned nowhere, moreover, my tests with latest versions of Firefox and Chrome has shown that it's in fact an incorrect behaviour and is not considered a closing tag by modern browsers.

Reply:
https://html.spec.whatwg.org/#comment-end-state is the section of the specs I was looking for.  It does indeed mention the `!` but not the spaces, so updating the code accordingly sounds good to me.

Do you want to add tests to check these (`-->`, `--!>`, `-- >`, `--x>`, `--->`, etc.) cases?

Reply:
> https://html.spec.whatwg.org/#comment-end-state is the section of the specs I was looking for. It does indeed mention the `!` but not the spaces, so updating the code accordingly sounds good to me.
> 
> Do you want to add tests to check these (`-->`, `--!>`, `-- >`, `--x>`, `--->`, etc.) cases?

I am thinking about improving the solution to even include `<!-->`, unexpected EOF and similar other test cases (that were mentioned in a [similar PR](https://github.com/ScintillaOrg/lexilla/issues/232)), but at the moment, unfortunately, I am lacking time to work on this PR. Hopefully, in the week (or at the weekend at worst) I can add the test cases and change a few other parts of the code to handle even wider variety of edge cases.